pallets/identity/src/auth.rs

```diff
@@ -112,6 +125,9 @@ impl<T: Config> Module<T> {
     ) {
         <Authorizations<T>>::remove(target, auth_id);
         <AuthorizationsGiven<T>>::remove(authorizer, auth_id);
+        NumberOfGivenAuths::mutate(authorizer, |number_of_given_auths| {
+            *number_of_given_auths = number_of_given_auths.saturating_sub(1);
+        });
```

Comment:
There are a few more places where an authorization is removed and we will need to decrease the `NumberOfGivenAuths`.